### PR TITLE
Supabase integration

### PR DESCRIPTION
**Pull request by SashankBalusu**
[//]: # "Feel free to customize this template to your project's vibes"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- content is populated via database

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

Screenshot 2024-10-09 at 11 50 59 PM

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

- component is viewable in view-post folder
- code is present in components/Post
- Can be viewed all at  once

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'
https://stackoverflow.com/questions/58709934/react-usestate-hooks-error-argument-of-type-xxx-is-not-assignable-to-paramete
https://stackoverflow.com/questions/63520629/how-to-fix-error-argument-of-type-undefined-is-not-assignable-to-parameter-of-ty
https://www.geeksforgeeks.org/typescript-array-map-method/
https://github.com/orgs/supabase/discussions/3218
https://stackoverflow.com/questions/66509263/how-to-use-an-async-database-call-to-set-a-variable-with-usestate-and-useeffec
https://react.dev/reference/react/useState
https://stackoverflow.com/questions/53650468/set-types-on-usestate-react-hook-with-typescript
https://supabase.com/docs/reference/javascript/eq
https://www.geeksforgeeks.org/typescript-array-map-method/#

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @insert pl github username here

### CODE REVIEW

File: components/Post/Post.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        ({
+          uuid: val.uuid as UUID,
+          created_at: val.created_at as string,
+          description: val.description as string,
+          username: val.username as string,
+          image_url: val.image_url as string,
+          like_count: val.like_count as number,
+        })!,
+    );
+    return format;
```
Comment (ronniebeggs) [inline components/Post/Post.tsx:44]: Given that we only want to display one post on screen at once, it may make more sense to store a single Post item state, rather than a full array of posts.

File: components/Post/Post.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                  style={{
+                    width: '-webkit-fill-available',
+                    borderRadius: '8px',
+                    marginTop: '2%',
+                  }}
+                  src={post.image_url}
+                ></img>
+                <div className={styles.postInteractions}>
+                  <HeartIcon size={post.like_count} />
+                  <p>256 Likes</p>
```
Comment (ronniebeggs) [inline components/Post/Post.tsx:84]: Like count is also stored in our database!

File: components/Post/Post.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        <p>Loading...</p>
+      ) : (
+        postData.map(post => (
+          <>
+            <div className={styles.content}>
+              <div className={styles.post}>
+                <div className={styles.headingInfo}>
+                  <ProfileIcon size={24} />
+                  <p>{post.username}</p>
+                  <p>{post.created_at}</p>
```
Comment (ronniebeggs) [inline components/Post/Post.tsx:71]: Consider using the built-in `Date` class for formatting date strings.

**Review by ronniebeggs — CHANGES_REQUESTED** [review]
Looks good to me! I left comments on a few minor things, but you built the core part of the assignment. Good job!
